packages/typography/src/elements/CodeBlock.tsx

```diff
@@ -28,13 +29,46 @@ interface IToken {
  */
 export const CodeBlock = React.forwardRef<HTMLPreElement, ICodeBlockProps>(
   (
-    { children, containerProps, highlightLines, isLight, isNumbered, language, size, ...other },
+    {
+      children,
+      containerProps,
+      highlightLines,
+      isLight,
+      isNumbered,
+      language = 'tsx',
+      size = 'medium',
+      ...other
+    },
     ref
   ) => {
     const containerRef = useRef<HTMLDivElement>(null);
     const code = (Array.isArray(children) ? children[0] : children) as string;
     const dependency = useMemo(() => [size, children], [size, children]);
     const containerTabIndex = useScrollRegion({ containerRef, dependency });
+    const [isPrismImported, setIsPrismImported] = useState(false);
+    const win = useWindow();
+
+    const importPrism = useCallback(async () => {
+      // TODO remove `importPrism` if/when `prismJS` releases v2 ESM
+      // https://github.com/orgs/PrismJS/discussions/3531
+      if (win && !isPrismImported) {
+        (win as any).Prism = Prism;
+
+        await Promise.all([
+          import('prismjs/components/prism-diff' as any),
+          import('prismjs/components/prism-json' as any)
+        ]);
+
+        setIsPrismImported(true);
+      }
+    }, [win, isPrismImported]);
+
+    useLayoutEffect(() => {
+      // Import languages missing from the vendored `Prism` using a variant of
+      // https://github.com/FormidableLabs/prism-react-renderer?tab=readme-ov-file#custom-language-support
+      // that is compatible with both React and Jest
+      importPrism();
+    }, [importPrism]);
```

Comment:
All of this goes away if/when PrismJS releases v2 with support for ESM imports 🙄